utilities/test_suite/HIP/testAllScript.sh

```diff
@@ -0,0 +1,292 @@
+
+# Fill with default values if all arguments are not given by user
+CASE_MIN=0
+CASE_MAX=86
+if (( "$#" < 4 )); then
+    SRC_FOLDER_1="$DEFAULT_SRC_FOLDER_1"
+    SRC_FOLDER_2="$DEFAULT_SRC_FOLDER_2"
+    PROFILING_OPTION="0"
+    TEST_TYPE="0"
+    QA_MODE="0"
+    DECODER_TYPE="0"
+    NUM_ITERATIONS="1"
+    CASE_LIST=()
+    for ((case=$CASE_MIN;case<=$CASE_MAX;case++))
+    do
+        CASE_LIST+=("$case")
+    done
+else
+    SRC_FOLDER_1="$1"
+    SRC_FOLDER_2="$2"
+    TEST_TYPE="$3"
+    NUM_ITERATIONS="$4"
+    PROFILING_OPTION="$5"
+    QA_MODE="$6"
+    DECODER_TYPE="$7"
+    CASE_LIST="${@:8}"
+fi
+
+if [[ $TEST_TYPE -ne 0 ]] && [[ $TEST_TYPE -ne 1 ]]; then
+    echo "Invalid TEST_TYPE specified. TEST_TYPE should be 0/1 (0 = Unittests / 1 = Performancetests)"
+    exit
+fi
+
+for case in $CASE_LIST; do
+    if [[ $case -lt 0 || $case -gt 86 ]]; then
+        echo "The case# must be in the 0:86 range!"
+    fi
+done
+
+# <<<<<<<<<<<<<< CREATE OUTPUT FOLDERS BASED ON TEST TYPE>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+if [ $TEST_TYPE -eq 0 ]; then
+    printf "\nRunning Unittests...\n"
+    mkdir "$cwd/../OUTPUT_IMAGES_HIP_NEW"
+    DEFAULT_DST_FOLDER="$cwd/../OUTPUT_IMAGES_HIP_NEW"
+elif [ $TEST_TYPE -eq 1 ]; then
+    printf "\nRunning Performance tests...\n"
+    rm -rvf "$cwd/../OUTPUT_PERFORMANCE_LOGS_HIP_NEW"
+    mkdir "$cwd/../OUTPUT_PERFORMANCE_LOGS_HIP_NEW"
+    DEFAULT_DST_FOLDER="$cwd/../OUTPUT_PERFORMANCE_LOGS_HIP_NEW"
+    LOGGING_FOLDER="$cwd/../OUTPUT_PERFORMANCE_LOGS_HIP_NEW"
+fi
+DST_FOLDER="$DEFAULT_DST_FOLDER"
+
+# <<<<<<<<<<<<<< EXECUTION OF ALL FUNCTIONALITIES (NEED NOT CHANGE) >>>>>>>>>>>>>>
+
+directory_name_generator() {
+
+    AFFINITY=$1
+    TYPE=$2
+    CASE=$3
+
+    if [[ "$case" -lt 5 ]] || [ "$case" -eq 13 ] || [ "$case" -eq 36 ]
+    then
+        FUNCTIONALITY_GROUP="color_augmentations"
+    elif [[ "$case" -eq 8 ]] || [ "$case" -eq 30 ] || [ "$case" -eq 83 ] || [ "$case" -eq 84 ]
+    then
+        FUNCTIONALITY_GROUP="effects_augmentations"
+    elif [[ "$case" -lt 40 ]]
+    then
+        FUNCTIONALITY_GROUP="geometric_augmentations"
+    elif [[ "$case" -lt 42 ]]
+    then
+        FUNCTIONALITY_GROUP="morphological_operations"
+    elif [[ "$case" -eq 49 ]]
+    then
+        FUNCTIONALITY_GROUP="filter_augmentations"
+    elif [[ "$case" -lt 86 ]]
+    then
+        FUNCTIONALITY_GROUP="data_exchange_operations"
+    else
+        FUNCTIONALITY_GROUP="miscellaneous"
+    fi
+
+    DST_FOLDER_TEMP="$DST_FOLDER""/rpp_""$AFFINITY""_""$TYPE""_""$FUNCTIONALITY_GROUP"
+}
+
+rm -rvf "$DST_FOLDER"/*
+shopt -s extglob
+mkdir build
+cd build
+rm -rvf ./*
+cmake ..
+make -j16
+
+if [[ $TEST_TYPE -eq 1 ]] && [[ $PROFILING_OPTION -eq 1 ]]; then
+    mkdir "$DST_FOLDER/Tensor_PKD3"
+    mkdir "$DST_FOLDER/Tensor_PLN1"
+    mkdir "$DST_FOLDER/Tensor_PLN3"
+fi
+
+# <<<<<<<<<<<<<< EXECUTION OF ALL FUNCTIONALITIES (NEED NOT CHANGE) >>>>>>>>>>>>>>
```

Comment:
I think this comment is repeated twice. Lets also add a few more comments on this file as to what each code segment does

Reply:
Done